### PR TITLE
feat(integrations): first-party integration analytics

### PR DESCRIPTION
**Pull request by scefali**
This PR adds analytics for the following events for first party integrations:

- integrations.installation_start
- integrations.installation_complete
- integrations.details_viewed
- integrations.uninstall_clicked
- integrations.uninstall_completed

And an event for the integrations index page:
- integrations.index_viewed

It also makes `org` an optional parameter to `trackIntegrationEvent`. This is because there are many instances of `AddIntegration` and it requires many changes to ensure the organization is always passed to the component. I was not able to use the `withOrganization` HoC because for some unknown reason, the organization context disappears when the popup opens. I am not 100% sure that we will need the org for the `installation_start` and `installation_complete` events so doing all that refactoring now is not worth it. But for the main installation flow, the org should be included as an argument.

Still need to add more analytics for Sentry Apps and plugins in future PRs.

Also need to add a Reload PR for the new events.

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/integrationUtil.tsx [hunk 1]
```diff
@@ -45,17 +67,25 @@ type IntegrationsEventParams = (MultipleIntegrationsEvent | SingleIntegrationEve
...
+  org?: Organization, //we should pass in org whenever we can but not every place guarantees this
   options?: {startSession: boolean}
 ) => {
   const {startSession} = options || {};
   const sessionId = startSession ? startAnalyticsSession() : getAnalyticsSessionId();
   const fullParams = {
     analytics_session_id: sessionId,
-    organization_id: org.id,
-    role: org.role,
+    organization_id: org && org.id,
```
Comment (NisanthanNanthakumar) [inline src/sentry/static/sentry/app/utils/integrationUtil.tsx]: Can we use optional chaining on line 77 & 78?

Comment (scefali) [reply src/sentry/static/sentry/app/utils/integrationUtil.tsx]: @NisanthanNanthakumar When this PR started, the answer was no. But now we can :)

Comment (scefali) [reply src/sentry/static/sentry/app/utils/integrationUtil.tsx]: @NisanthanNanthakumar Just kidding...the optional chaining PR has not been merged yet. If it gets merged before this PR gets approved I will add the change.

Comment (NisanthanNanthakumar) [reply src/sentry/static/sentry/app/utils/integrationUtil.tsx]: sounds good

**Review by MeredithAnya — CHANGES_REQUESTED** [review]
just one small fix

**Comment by MeredithAnya** [comment]
Not sure if this was discussed elsewhere but wondering why we need the whole Organization object passed through since it seems like we are only using the organization id? Also not sure if it's super necessary to change, just was wondering if there is a reason or if I missed something.

**Comment by scefali** [comment]
@MeredithAnya We are also passing in the `role`. I thought it made sense to pass in the org as one argument instead of having to pass in two different arguments derived from the org.

**Comment by MeredithAnya** [comment]
> @MeredithAnya We are also passing in the `role`. I thought it made sense to pass in the org as one argument instead of having to pass in two different arguments derived from the org.

missed the role, sounds good

File: src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegration.tsx [hunk 2]
```diff
@@ -146,7 +173,7 @@ export default class InstalledIntegration extends React.Component<Props> {
                     borderless
                     icon="icon-settings"
                     disabled={!this.hasConfiguration() || !hasAccess}
-                    to={`/settings/${orgId}/integrations/${provider.key}/${integration.id}/`}
+                    to={`/settings/${organization.id}/integrations/${provider.key}/${integration.id}/`}
```
[the remark this answers is not in the document]
Comment (scefali) [reply src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegration.tsx]: @MeredithAnya You are right! Good catch.

**Review by MeredithAnya — APPROVED** [review]

**Pull request merged**